### PR TITLE
GET /ideas?filter[commentedBy]='username0,username1,username2' 

### PR DESCRIPTION
**Pull request by agatatalita**
It took about 1h, but most of it was AQL.

### CODE REVIEW

File: models/idea/index.js [hunk 1]
```diff
@@ -216,6 +216,63 @@ class Idea extends Model {
...
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findCommentedBy(commentedBy, { offset, limit }) {
+
+    const query = `
+      FOR user IN users 
+      FILTER user.username IN @commentedBy
+        FOR comment IN comments 
+        FILTER comment.creator == user._id 
+        AND SUBSTRING(comment.primary, 0, 5) == 'ideas'
```
Comment (mrkvon) [inline models/idea/index.js]: This should be replaced by more appropriate [`IS_SAME_COLLECTION`](https://docs.arangodb.com/3.3/AQL/Functions/Document.html#issamecollection]) or [`PARSE_IDENTIFIER`](https://docs.arangodb.com/3.3/AQL/Functions/Document.html#parseidentifier) functions.

File: models/idea/index.js [hunk 2]
```diff
@@ -216,6 +216,63 @@ class Idea extends Model {
...
+      FILTER user.username IN @commentedBy
+        FOR comment IN comments 
+        FILTER comment.creator == user._id 
+        AND SUBSTRING(comment.primary, 0, 5) == 'ideas' 
+          FOR idea IN ideas 
+          FILTER idea._id == comment.primary
+          // sort from newest
+          SORT idea._id DESC, idea.created ASC
+          LIMIT @offset, @limit
+      RETURN DISTINCT idea`;
```
Comment (mrkvon) [inline models/idea/index.js]: Nice alternative to `COLLECT` (:+1:), although according to [docs](https://docs.arangodb.com/3.3/AQL/Examples/Grouping.html#ensuring-uniqueness), `RETURN DISTINCT` has undefined ordering. That might be the source of the Travis-CI-test fails. Will need to be replaced with `COLLECT` after all.

File: models/idea/index.js [hunk 3]
```diff
@@ -245,6 +245,35 @@ class Idea extends Model {
...
+      FOR user IN users 
+      FILTER user.username IN @commentedBy
+        FOR comment IN comments 
+        FILTER comment.creator == user._id 
+        AND IS_SAME_COLLECTION('ideas', comment.primary)
+          FOR idea IN ideas 
+          FILTER idea._id == comment.primary
+          COLLECT i = idea
+          // sort from newest
+          SORT i._id DESC, i.created ASC
```
Comment (mrkvon) [inline models/idea/index.js]: Is `SORT i._id DESC` necessary here?

File: test/ideas.list.js [hunk 4]
```diff
@@ -542,4 +542,146 @@ describe('read lists of ideas', () => {
...
+        user3,
+        user4;
+    // create and save testing data
+    beforeEach(async () => {
+      const data = {
+        users: 6,
+        tags: 6,
+        verifiedUsers: [0, 1, 2, 3, 4],
+        ideas: [[{}, 0], [{}, 0],[{}, 1],[{}, 2],[{}, 2],[{}, 2],[{}, 3]],
+        userTag: [
```
Comment (mrkvon) [inline test/ideas.list.js]: Is `userTag` and `ideaTags` necessary here?

Comment (agatatalita) [reply test/ideas.list.js]: I thought we stated we are leaving it if it not disturbing.

File: models/idea/index.js (hunk 3 above)
Comment (agatatalita) [reply models/idea/index.js]: it is not

File: models/idea/index.js [hunk 5]
```diff
@@ -266,7 +266,7 @@ class Idea extends Model {
           FILTER idea._id == comment.primary
           COLLECT i = idea
           // sort from newest
-          SORT i.created ASC
+          SORT i.created DESC
```
Comment (mrkvon) [inline models/idea/index.js]: Ah, cool. :+1:

**Pull request merged**